src/components/search-date-picker.jsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import 'react-dates/initialize';
+import { DateRangePicker } from 'react-dates';
+import 'react-dates/lib/css/_datepicker.css';
+import moment from 'moment';
+import { isInclusivelyBeforeDay } from 'react-dates';
+
+/**
+ * @typedef SearchDatePicker
+ * @property {Function} onDateSearch
+ */
+
+/**
+ * Renders date picker which allows startDate, endDate or both to be selected
+ * Once a date is selected, onDateSearch function is called with startDate and endDate values
+ * Dates can also be cleared out after selected
+ *
+ * @class SearchDatePicker
+ * @extends {React.Component}
+ * @param {SearchDatePickerProps} props
+ */
+export default class SearchDatePicker extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      startDate: null,
+      endDate: null,
+      focusedInput: null,
+    };
+    this.onDatesChange = this.onDatesChange.bind(this);
+  }
+
+  componentDidUpdate(previousProps, previousState) {
+    if (previousState.startDate !== this.state.startDate || previousState.endDate !== this.state.endDate) {
+      this.props.onDateSearch({startDate: this.state.startDate, endDate: this.state.endDate});
+    }
+  }
+
+  onDatesChange({ startDate, endDate }) {
+    this.setState({ startDate, endDate });
+  }
+
+  render() {
+    return (
+      <DateRangePicker
+        startDateId="startDate"
+        endDateId="endDate"
```

Comment:
This is OK since we only ever use this component once, but this is probably not so great in the abstract case (e.g. maybe some future animated transition might put more than one of these in the DOM). It might be good to set some unique value in the constructor that we can add to these:

```js
  constructor (props) {
    // ...other constructor work...

    this.id = Math.random().toString().slice(2);
  }

  render () {
    return (
      <DateRangePicker
        startDateId={startDate + this.id}
        endDateId={endDate + this.id}
        ...other props...
      />
    )
  }
```

Or maybe this component could accept an `id` in its props, so you don’t need to gin one up in the constructor:

```js
  render () {
    return (
      <DateRangePicker
        startDateId={startDate + (this.props.id || '')}
        endDateId={endDate + (this.props.id || '')}
        ...other props...
      />
    )
  }
```

Reply:
You might want to combine this with what I was thinking about the tests here: https://github.com/edgi-govdata-archiving/web-monitoring-ui/pull/422#discussion_r344904475